projects/plugins/boost/app/modules/cache/pre-wordpress/Boost_Cache_Utils.php

```diff
@@ -9,11 +9,11 @@ class Boost_Cache_Utils {
 	/**
 	 * Recursively delete a directory.
 	 * @param string $dir - The directory to delete.
-	 * @param bool   $recurse - If false, only delete the files in the directory, do not recurse into subdirectories.
+	 * @param bool   $filter - The filter to use. '*' to delete all files in the given directory. '*\/' to delete everything in the given directory, recursively. '\/**\/' to delete a single file or directory in the given directory.
```

Comment:
The `glob` pattern filter idea sounds good on paper, but actually seeing it used I wonder if we can simplify it. I think you had an idea of an array before, perhaps we should do that instead?

Another idea would be to predefine a list of constants that we could use as flags for how to filter

```php
define("JBCACHE_ALL", 1);
define("JBCACHE_FILES", 2);
```

And then accept either an integer (the flag) or an array like:

```php
delete_directory($dir, [ 'hash' => $hash, 'type' => JBCACHE_FILES ] );
```

and

```php
delete_directory($dir, JBCACHE_ALL );
```

Reply:
Oh, even simpler:

```php
delete_directory($dir, JBCACHE_FILES, $hash );
```

and 

```php
delete_directory($dir, JBCACHE_ALL );
```

Reply:
Seems like a good idea to use those constants. Just added them.